[2025-10-17T20:20:02.710+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-10-17T20:20:02.745+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: flood_prediction_pipeline.fetch_weather_data scheduled__2025-10-17T20:15:00+00:00 [queued]>
[2025-10-17T20:20:02.755+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: flood_prediction_pipeline.fetch_weather_data scheduled__2025-10-17T20:15:00+00:00 [queued]>
[2025-10-17T20:20:02.757+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-10-17T20:20:02.777+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): fetch_weather_data> on 2025-10-17 20:15:00+00:00
[2025-10-17T20:20:02.793+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=3150) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-10-17T20:20:02.791+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'flood_prediction_pipeline', 'fetch_weather_data', 'scheduled__2025-10-17T20:15:00+00:00', '--job-id', '74', '--raw', '--subdir', 'DAGS_FOLDER/flood_prediction_dag.py', '--cfg-path', '/tmp/tmpk4qadoba']
[2025-10-17T20:20:02.794+0000] {standard_task_runner.py:63} INFO - Started process 3153 to run task
[2025-10-17T20:20:02.795+0000] {standard_task_runner.py:91} INFO - Job 74: Subtask fetch_weather_data
[2025-10-17T20:20:02.866+0000] {task_command.py:426} INFO - Running <TaskInstance: flood_prediction_pipeline.fetch_weather_data scheduled__2025-10-17T20:15:00+00:00 [running]> on host fcca755cbcb6
[2025-10-17T20:20:03.004+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='flood_prediction_pipeline' AIRFLOW_CTX_TASK_ID='fetch_weather_data' AIRFLOW_CTX_EXECUTION_DATE='2025-10-17T20:15:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-10-17T20:15:00+00:00'
[2025-10-17T20:20:03.007+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-10-17T20:20:03.038+0000] {logging_mixin.py:188} INFO - âœ… Weather data fetched: {'temp': 31.07014113764779, 'humidity': 85.92117994788715, 'pressure': 981.7779807575752, 'wind_speed': 17.4183703528119, 'rain_1h': 0.3978499964899762, 'rain_3h': 52.0521536973792, 'rain_6h': 139.50284038481857, 'rain_24h': 283.6294656597196, 'rain_intensity': 2.0142874914119933, 'temp_delta': 0.0, 'humidity_delta': 0.0, 'hour': 20, 'minute': 20, 'second': 3, 'day': 17, 'month': 10, 'year': 2025}
[2025-10-17T20:20:03.038+0000] {python.py:237} INFO - Done. Returned value was: {'temp': 31.07014113764779, 'humidity': 85.92117994788715, 'pressure': 981.7779807575752, 'wind_speed': 17.4183703528119, 'rain_1h': 0.3978499964899762, 'rain_3h': 52.0521536973792, 'rain_6h': 139.50284038481857, 'rain_24h': 283.6294656597196, 'rain_intensity': 2.0142874914119933, 'temp_delta': 0.0, 'humidity_delta': 0.0, 'hour': 20, 'minute': 20, 'second': 3, 'day': 17, 'month': 10, 'year': 2025}
[2025-10-17T20:20:03.039+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-10-17T20:20:03.069+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=flood_prediction_pipeline, task_id=fetch_weather_data, run_id=scheduled__2025-10-17T20:15:00+00:00, execution_date=20251017T201500, start_date=20251017T202002, end_date=20251017T202003
[2025-10-17T20:20:03.129+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-10-17T20:20:03.153+0000] {taskinstance.py:3498} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-10-17T20:20:03.158+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
